Source/Evolve/WarpXComputeDt.cpp

```diff
@@ -58,16 +63,11 @@ WarpX::ComputeDt ()
 
     if (m_const_dt.has_value()) {
         deltat = m_const_dt.value();
-    } else if (electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD) {
+    } else if (electrostatic_solver_id  != ElectrostaticSolverAlgo::None ||
+               electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD) {
         // Computation of dt for spectral algorithm
         // (determined by the minimum cell size in all directions)
-#if defined(WARPX_DIM_1D_Z)
-        deltat = cfl * dx[0] / PhysConst::c;
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-        deltat = cfl * std::min(dx[0], dx[1]) / PhysConst::c;
-#else
-        deltat = cfl * std::min(dx[0], std::min(dx[1], dx[2])) / PhysConst::c;
-#endif
+        deltat = cfl / PhysConst::c * minDim(dx);
```

Comment:
Since this is using `c`, this `dt` for the first time step would typically be much smaller than needed. Could `UpdateDtFromParticleSpeeds` be called from `WarpX::InitData` after the particles have been initialized to give an appropriate `dt` for the first step?

Reply:
My thinking was that in most real simulations, you use enough timesteps that a small first timestep really won't matter, and that I wanted to have all of the timestep-setting logic in one place in the code. Would it break much to instead move the call to `ComputeDt` to after the particles have been initialized?

Reply:
I was thinking the same thing. Though the `InitPML` in `InitFromScratch` look likes it depend on `dt`.

Reply:
I am testing moving the timestep update to the start of the `Evolve` loop. Previously we would actually get two very small timesteps as `dt_next` wouldn't be updated until the end of the first iteration. A new problem has arisen though -- what to do when particles are at rest? The `ElectrostaticSphere` case starts with particles with zero velocity, so the computed adaptive timestep is infinite. I'm testing just setting it to a maximum value for now but that isn't especially principled.

Reply:
It might be good to add a user-specified maximum timestep as a fix for this